src/stores/snapshot-slice.ts

```diff
@@ -0,0 +1,205 @@
+import crypto from 'crypto';
+import { createSlice, createAsyncThunk, PayloadAction, createSelector } from '@reduxjs/toolkit';
+import { __, sprintf } from '@wordpress/i18n';
+import { CreateLoggerAction } from 'cli/commands/preview/logger-actions';
+import { LIMIT_OF_ZIP_SITES_PER_USER } from 'src/constants';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import { RootState, store } from 'src/stores/index';
+
+export type SnapshotOperation = {
+	detail: string;
+	error: string | null;
+	progress: number;
+	snapshotName?: string;
+	snapshotUrl?: string;
+	siteId: string;
+	status: 'pending' | 'fulfilled' | 'rejected';
+	type: 'create' | 'update' | 'delete';
+};
+
+type SnapshotState = {
+	operations: Record< crypto.UUID, SnapshotOperation >;
+	snapshotProgress: number;
+	snapshots: Snapshot[];
+	snapshotQuota: number;
+};
+
+const getInitialState = (): SnapshotState => {
+	return {
+		operations: {},
+		snapshotProgress: 0,
+		snapshots: [],
+		snapshotQuota: LIMIT_OF_ZIP_SITES_PER_USER,
+	};
+};
+
+const getSnapshots = createAsyncThunk( 'snapshot/getSnapshots', async () => {
+	return await getIpcApi().getSnapshots();
+} );
+
+const createSnapshot = createAsyncThunk(
+	'snapshot/createSnapshot',
+	async ( { siteId, siteFolder }: { siteId: string; siteFolder: string } ) => {
+		const { operationId } = await getIpcApi().createSnapshot( siteFolder );
+		return { operationId, siteId };
+	}
+);
+
+const snapshotSlice = createSlice( {
+	name: 'snapshot',
+	initialState: getInitialState(),
+	reducers: {
+		updateOperation: (
+			state,
+			action: PayloadAction< { operationId: crypto.UUID; operation: Partial< SnapshotOperation > } >
+		) => {
+			Object.assign( state.operations[ action.payload.operationId ], action.payload.operation );
+		},
+	},
+	extraReducers: ( builder ) => {
+		builder
+			.addCase( getSnapshots.fulfilled, ( state, action ) => {
+				state.snapshots = action.payload;
+			} )
+			.addCase( createSnapshot.fulfilled, ( state, action ) => {
+				state.operations[ action.payload.operationId ] = {
+					detail: __( 'Creating archive...' ),
+					error: null,
+					progress: 0,
+					siteId: action.payload.siteId,
+					status: 'pending',
+					type: 'create',
+				};
+			} );
+	},
+	selectors: {
+		selectSnapshots: ( state ) => state.snapshots,
+		selectSnapshotsCount: ( state ) => state.snapshots.length,
+		selectActiveOperationForSite: ( state, siteId: string ) =>
+			Object.values( state.operations )
+				.filter( ( operation ) => operation.status === 'pending' )
+				.find( ( operation ) => operation.siteId === siteId ),
```

Comment:
Can we go this way?
```suggestion
				.filter( ( operation ) => operation.status === 'pending' && operation.siteId === siteId )
```

Reply:
I see that you already did it in the follow up PR https://github.com/Automattic/studio/pull/1223/files#diff-e7f3116a80918052c6f2b0b4cc2e22b2f8dc46e44d97249c6d49522bc2bf3b2cR127-R131